### ISSUE TITLE
Registration Call Back Function Does Not Get Called on iOS

### ISSUE THREAD

**Issue by jiankai82** [phonegap/phonegap-plugin-push#947]
### Expected Behaviour
push.on('registration', myCallBackFunction)
myCallBackFunction gets called after registration completes.
### Actual Behaviour
myCallBackFunction does not get called after registration completes.
### Reproduce Scenario (including but not limited to)
Start up the app
#### Steps to Reproduce 
Startup the app
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 9.3.2
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Apple
#### Cordova CLI version and cordova platform version
cordova CLI version 5.3.3
cordova platform version iOS 3.9.2
Note: I had to change plugin.xml to set the minimum Cordova iOS platform version to 3.9.2 to work with my app.
#### Plugin version

    cordova plugin version | grep phonegap-plugin-push 1.6.3
Note: I had to change plugin.xml to set the minimum Cordova iOS platform version to 3.9.2 to work with my app.
#### Sample Code that illustrates the problem
var push = PushNotification.init({
    ios: {
        alert: "true",
        badge: "true",
        sound: "true"
    }
});
push.on('registration', myCallBackFunction);

myCallBackFunction does not get called after registration completes.
I debugged the plugin native code (objective C) in xCode.  I saw the registration is successful.  However, the JavaScript code that registers the registration call back function does not get called until after the registration completes which is why myCallBackFunction does not get called. i.e. push.on('registration', myCallBackFunction) does not get called until after (void)registerWithToken:(NSString*)token; returns.

Anyone has any idea of why this is happening?

**Comment by fredgalvao** [comment]
I can't say for sure this is related to your issue, but:

> Note: I had to change plugin.xml to set the minimum Cordova iOS platform version to 3.9.2 to work with my app.

You modified the source code of the plugin to your needs, and by doing that you probably skipped our assumption of what works and what doesn't. **That version requirement is there for a reason**, and if you can't satisfy it, you have the following options:
* update your project so that it satisfies this plugin's requirements
* don't use the plugin at all
* use a previous version of this plugin that supports cordova-ios 3.9.x
* modify it and take responsibility of any problems occurring with your custom version of the plugin

**Comment by macdonst** [comment]
@jiankai82 as @fredgalvao said there is a reason why we specify cordova-ios higher than 4 for later versions of the plugin. The fact you changed that in your plugin makes this issue kinda invalid.

However, it sounds like you are getting the registration event once (void)registerWithToken:(NSString*)token; returns which is proper behaviour. Can you confirm your registration event handler eventually gets called?

**Comment by jiankai82** [comment]
Thanks guys for your input.  I understand modifying the plugin source code voids the warranty so I am on my own.  I believe the only reason the plugin requires Cordova-iOS 4.0.0 or higher is because of bit code.  Can anyone confirm this is the case?  I downgraded it to 3.9.2 since we do not use bit code in our app.  I hope this will not cause any problem.

Anyways, regarding my problem, I got it working.  My original implementation is to call var push = PushNotification.init(...) in a separate function.  I then pass the push object as a parameter to another function which calls push.on('registration',...), push.on('error',...) and push.on('notification',...).  This implementation **works fine for Android** but not for iOS.  The registration event handler never gets called for iOS.  I changed my implementation by moving var push = PushNotification.init(...) to the same function as the handlers, just like what the plugin example is doing.  This worked.  The registration event handler gets called and all is good.  I just don't understand why this works but not my original implementation, especially my original implementation works for Android but not for iOS.  It'd be great if someone can shed some light on this.  Thanks.
